sympy/core/expr.py

```diff
@@ -1020,6 +1020,17 @@ def conjugate(self):
         from sympy.functions.elementary.complexes import conjugate as c
         return c(self)
 
+    def dir(self, x, cdir):
+        minexp = S.Zero
+        arg = self
```

Comment:
It should be checked somewhere, maybe here, that `arg` is not 0 to begin with.

Reply:
So something like:
```
if arg.is_zero:
    return
```

Reply:
How about retuning 0 which is the default?

Reply:
should work fine.